lib/ini.js

```diff
@@ -104,16 +123,30 @@ function splitSections (str, separator) {
 }
 
 const decode = (str, opt = {}) => {
+  // The `typeof` check is required because accessing the `process` directly fails on browsers.
+  /* istanbul ignore next */
+  opt.platform = opt.platform || (typeof process !== 'undefined' && process.platform)
+  /* istanbul ignore next */
+  const eol = opt.platform === 'win32' ? '\r\n' : '\n'
   opt.bracketedArray = opt.bracketedArray !== false
   const out = Object.create(null)
   let p = out
   let section = null
+  const preserveComments = opt.preserveComments || false
+  const commentsKey = opt.commentsKey || 'comments'
+  const commentsDictionary = {}
+  let lineCommentArray = []
+  const commentsRegEx = /^[#;]+[^\n]+$/
```

Comment:
Since we're not doing any grouping or matching, do we really need the `+` after matching `#` or `;`?  They both still would match `[^\n]` anyways.

Also, since the line is already the result of `str.split(/[\r\n]+/g)` I don't understand how it could contain `\n` in the first place.

Reply:
I guess I was over thinking the solution. I will push an update. Thanks for the feedback.